src/openms/include/OpenMS/CHEMISTRY/ISOTOPEDISTRIBUTION/CoarseIsotopePatternGenerator.h

```diff
@@ -154,6 +154,24 @@ namespace OpenMS
     */
     IsotopeDistribution estimateFromPeptideWeightAndS(double average_weight, UInt S);
 
+    /**
+       @brief roughly approximate peptide IsotopeDistribution from monoisotopic weight using Poisson distribution.
+       m/z values are somewhat arbitrary. Foundation from: Bellew et al, https://dx.doi.org/10.1093/bioinformatics/btl276
+
+       @param average_weight: m/z of monoisotopic peak (with charge = 1) to approximate the distribution of intensities for
+       @param num_peaks: How many peaks should be generated (independent of this->max_isotope)
+    */
+    static IsotopeDistribution approximateFromPeptideWeight(double mass, int num_peaks = 20);
+
+    /**
+       @brief roughly approximate intensity distribution of peptidic isotope patterns from monoisotopic weight using Poisson distribution.
+       Foundation from: Bellew et al, https://dx.doi.org/10.1093/bioinformatics/btl276
+
+       @param average_weight: m/z of monoisotopic peak (with charge = 1) to approximate the distribution of intensities for
+       @param num_peaks: How many peaks should be generated (independent of this->max_isotope)
+    */
+    static std::vector<float> approximateIntensities(double mass, int num_peaks = 20);
```

Comment:
why not use double here to be more accurate? or a template argument so that a client could use a more accurate type?

Reply:
I think you mean because of a potential float overflow? Accuracy-wise, I think float is fine (natural abundances fluctuate a bit on earth anyway). 
@emilpaulitz do you know if the differences are caused by assuming a Poisson distribution (as opposed to calculating convolution of binomial distributions + convolutions) or by numeric inaccuracies?
Nevertheless, I think we could add a template here and use e.g. the double version in the deisotoper?

Reply:
@emilpaulitz does double make a noticeable speed difference? if not we could go with double instead I guess. Having a template is maybe a bit overkill here (also makes wrapping more difficult).

Reply:
I would suspect (tdb), that the approximation error is far greater than the loss of precision when comparing float to double.
But that is easy to check, I guess.

Reply:
The differences are definitely caused by assuming a poisson distribution, as estimateFromPeptideWeight (which returns IsotopeDistribution) also uses float for the intensites. In fact, the differences seem to start at the 3rd digit the latest, and if I recall correctly, float has 6 accurate digits. But switching to double does not seem to make a difference time-wise, so I did.